File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/idscp_core/idscp_server/ServerConnectionListener.java
package de.fhg.aisec.ids.idscp2.idscp_core.idscp_server;

import de.fhg.aisec.ids.idscp2.idscp_core.Idscp2Connection;

public interface ServerConnectionListener {

    void onConnectionCreated(Idscp2Connection connection);

    void onConnectionClose(Idscp2Connection connection);
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/idscp_core/api/idscp_server/ServerConnectionListener.kt
package de.fhg.aisec.ids.idscp2.idscp_core.api.idscp_server

import de.fhg.aisec.ids.idscp2.idscp_core.api.idscp_connection.Idscp2Connection

/**
 * A ServerConnectionListener interface for the Idscp2Server to get notification about
 * Idscp2Connection lifetimes in an isolated way
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
interface ServerConnectionListener<CC: Idscp2Connection> {
    fun onConnectionCreated(connection: CC)
    fun onConnectionClose(connection: CC)
}